app/forms/support_user_invite_form.rb

```diff
@@ -0,0 +1,51 @@
+class SupportUserInviteForm < ApplicationForm
+  FORM_PARAMS = %i[first_name last_name email].freeze
+
+  attr_accessor :email, :first_name, :last_name, :service
+
+  validate :validate_support_user
+  validates :service, presence: true
+
+  def invite!
+    save_support_user
+  end
+
+  def as_form_params
+    { "support_user" => slice(FORM_PARAMS) }
+  end
+
+  private
+
+  def support_user
+    @support_user ||=
+      begin
+        user = user_klass
+          .with_discarded
+          .discarded
+          .find_or_initialize_by(email:)
+        user.assign_attributes(first_name:, last_name:)
+        user
+      end
+  end
+
+  def save_support_user
+    ActiveRecord::Base.transaction do
+      support_user.undiscard! if support_user.discarded?
+      support_user.save!
+    end
+    SupportUser::Invite.call(support_user:)
+  end
```

Comment:
This breaks the pattern defined in our conventions docs. Side effects should be triggered outside of the Form Object.

> `save` should then be called in Controllers and [Service Objects](https://github.com/DFE-Digital/itt-mentor-services/blob/main/docs/conventions/service_objects.md).

https://github.com/DFE-Digital/itt-mentor-services/blob/main/docs/conventions/form_objects.md#activerecord-duck-typing